File: backend/src/main/java/com/project2/domain/chat/dto/ChatRoomResponseDTO.java
package com.project2.domain.chat.dto;

import java.util.UUID;

import com.project2.domain.chat.entity.ChatRoom;
import com.project2.domain.member.entity.Member;

import lombok.Getter;

@Getter
public class ChatRoomResponseDTO {
	private final UUID id;
	private final Member opponent;

	public ChatRoomResponseDTO(ChatRoom chatRoom, Long actorId) {
		this.id = chatRoom.getId();
		this.opponent = chatRoom.getMembers()
			.stream()
			.filter(member -> !member.getId().equals(actorId))
			.findFirst()
			.orElseThrow(() -> new IllegalStateException("ChatRoom must have exactly one opponent"));
	}

}

File: backend/src/main/kotlin/com/project3/domain/chat/dto/ChatRoomResponseDTO.kt
package com.project3.domain.chat.dto

import com.project3.domain.chat.entity.ChatRoom
import com.project3.domain.member.entity.Member
import java.util.*

data class ChatRoomResponseDTO(
        val id: UUID,
        val opponent: Member
) {
    companion object {
        @JvmStatic
        fun from(chatRoom: ChatRoom, actorId: Long): ChatRoomResponseDTO {
            val opponent = chatRoom.members
                    .find { it.id != actorId }
                    ?: throw IllegalStateException("ChatRoom must have exactly one opponent")
            val id = requireNotNull(chatRoom.id) { "ChatRoom.id must not be null" }

            return ChatRoomResponseDTO(
                    id = id,
                    opponent = opponent
            )
        }
    }
}